packages/flutter/lib/src/material/date_picker_theme.dart

```diff
@@ -163,6 +164,10 @@ class DatePickerThemeData with Diagnosticable {
   /// indicate that a day in the grid is focused, hovered, or pressed.
   final MaterialStateProperty<Color?>? dayOverlayColor;
 
+  /// Overrides the default shape used to paint the shape decoration of the
+  /// day labels in the grid of the date picker.
+  final MaterialStateProperty<OutlinedBorder?>? dayShape;
```

Comment:
Since this property only appears in the theme (which is OK) you should explain how it interacts with `todayBorder` and `todayForegroundColor`